plasma_framework/contracts/src/exits/payment/controllers/PaymentProcessInFlightExit.sol

```diff
@@ -0,0 +1,224 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "../PaymentExitDataModel.sol";
+import "../PaymentInFlightExitModelUtils.sol";
+import "../../../vaults/EthVault.sol";
+import "../../../vaults/Erc20Vault.sol";
+import "../../../framework/PlasmaFramework.sol";
+
+library PaymentProcessInFlightExit {
+    using PaymentInFlightExitModelUtils for PaymentExitDataModel.InFlightExit;
+
+    uint8 public constant MAX_INPUT_NUM = 4;
+    uint8 public constant MAX_OUTPUT_NUM = 4;
+
+    struct Controller {
+        PlasmaFramework framework;
+        EthVault ethVault;
+        Erc20Vault erc20Vault;
+    }
+
+    event InFlightExitOmitted(
+        uint192 indexed exitId,
+        address token
+    );
+
+    event InFlightExitOutputWithdrawn(
```

Comment:
I think that `outputIndex` is pretty misleading in the original design. It is `index + MAX_INPUT_SIZE` which in my opinion is sth really of implementation detail that I don't see why client should know.

I renamed it `Withdrawn` at the end since an input/output can be finalized without withdraw action. (like impossible to re-process, re-exit...) but I am okay to change this back to `...Finalized`.